In [30]:
pip install pandas


[notice] A new release of pip available: 22.3.1 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


## Задание 1.
1. Скачать данные по ссылке
https://www.kaggle.com/datasets/ionaskel/laptop-prices
2. Считать данные с помощью pandas
3. Вывести на экран первые 5 строк
4. Посмотреть на описание признаков и на их содержание

In [31]:
import pandas as pd
df = pd.read_csv('laptop_price.csv', encoding = 'windows-1251', sep = ',')
df.head()

laptop_ID Company      Product   TypeName  Inches  \
0          1   Apple  MacBook Pro  Ultrabook    13.3   
1          2   Apple  Macbook Air  Ultrabook    13.3   
2          3      HP       250 G6   Notebook    15.6   
3          4   Apple  MacBook Pro  Ultrabook    15.4   
4          5   Apple  MacBook Pro  Ultrabook    13.3   

                     ScreenResolution                         Cpu   Ram  \
0  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                            1440x900        Intel Core i5 1.8GHz   8GB   
2                   Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3  IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4  IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   

                Memory                           Gpu  OpSys  Weight  \
0            128GB SSD  Intel Iris Plus Graphics 640  macOS  1.37kg   
1  128GB Flash Storage        Intel HD Graphics 6000  macOS  1.34kg   
2            256GB SSD         Intel HD Graphics 620  No OS  1.86kg   
3            512GB SSD            AMD Radeon Pro 455  macOS  1.83kg   
4            256GB SSD  Intel Iris Plus Graphics 650  macOS  1.37kg   

   Price_euros  
0      1339.69  
1       898.94  
2       575.00  
3      2537.45  
4      1803.60

## Задание 2.
Проведите первичный анализ данных
1. Изучите типы данных
2. Найдите количество пропущенных ячеек в данных
3. Посчитайте основные статистики по всем признакам и поизучайте их

In [32]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         1303 non-null   int64  
 1   Company           1303 non-null   object 
 2   Product           1303 non-null   object 
 3   TypeName          1303 non-null   object 
 4   Inches            1303 non-null   float64
 5   ScreenResolution  1303 non-null   object 
 6   Cpu               1303 non-null   object 
 7   Ram               1303 non-null   object 
 8   Memory            1303 non-null   object 
 9   Gpu               1303 non-null   object 
 10  OpSys             1303 non-null   object 
 11  Weight            1303 non-null   object 
 12  Price_euros       1303 non-null   float64
dtypes: float64(2), int64(1), object(10)
memory usage: 132.5+ KB


In [33]:
df.describe()

laptop_ID       Inches  Price_euros
count  1303.000000  1303.000000  1303.000000
mean    660.155794    15.017191  1123.686992
std     381.172104     1.426304   699.009043
min       1.000000    10.100000   174.000000
25%     331.500000    14.000000   599.000000
50%     659.000000    15.600000   977.000000
75%     990.500000    15.600000  1487.880000
max    1320.000000    18.400000  6099.000000

In [34]:
df.describe(include=['object'])

Company Product  TypeName   ScreenResolution  \
count     1303    1303      1303               1303   
unique      19     618         6                 40   
top       Dell  XPS 13  Notebook  Full HD 1920x1080   
freq       297      30       727                507   

                               Cpu   Ram     Memory                    Gpu  \
count                         1303  1303       1303                   1303   
unique                         118     9         39                    110   
top     Intel Core i5 7200U 2.5GHz   8GB  256GB SSD  Intel HD Graphics 620   
freq                           190   619        412                    281   

             OpSys Weight  
count         1303   1303  
unique           9    179  
top     Windows 10  2.2kg  
freq          1072    121

## Задание 3.
Ответьте на несколько вопросов
Выведите все характеристики только по этому ноутбуку
Если таких ноутбуков несколько, то выводите их всех

### 3.1 Ноутбуков от какой компании больше всего в наборе данных?


In [35]:
df.value_counts('Company')

Company
Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Vero           4
Xiaomi         4
Google         3
Fujitsu        3
Chuwi          3
LG             3
Huawei         2
Name: count, dtype: int64

In [36]:
df.value_counts('Company').head(1)

Company
Dell    297
Name: count, dtype: int64

### 3.2 Какая минимальная и максимальная стоимость у ноутбуков в данных?


In [37]:
df['Price_euros'].max()

6099.0

In [38]:
df['Price_euros'].min()

174.0

### 3.3 Какой самый дорогой ноутбук в данных?


In [39]:
df[(df['Price_euros'] == df['Price_euros'].max()) | (df['Price_euros'] == df['Price_euros'].min())]

laptop_ID Company                           Product TypeName  Inches  \
196         200   Razer                         Blade Pro   Gaming    17.3   
1215       1233    Acer  C740-C9QX (3205U/2GB/32GB/Chrome  Netbook    11.6   

                         ScreenResolution  \
196   4K Ultra HD / Touchscreen 3840x2160   
1215                             1366x768   

                                       Cpu   Ram    Memory  \
196            Intel Core i7 7820HK 2.9GHz  32GB   1TB SSD   
1215  Intel Celeron Dual Core 3205U 1.5GHz   2GB  32GB SSD   

                          Gpu       OpSys  Weight  Price_euros  
196   Nvidia GeForce GTX 1080  Windows 10  3.49kg       6099.0  
1215        Intel HD Graphics   Chrome OS   1.3kg        174.0

## Задание 4.
Ответьте на несколько вопросов

### 4.1 Найдите ноутбуки с самой маленькой диагональю в данных?
Выведите только следующие характеристики:
1. Компания
2. Типа ноутбука
3. Диагональ
4. Стоимость
Если таких ноутбуков несколько, то выводите их всех

In [40]:
df[['Company', 'TypeName', 'Inches', 'Price_euros']]. sort_values('Inches').head(4)

Company            TypeName  Inches  Price_euros
1114  Lenovo  2 in 1 Convertible    10.1       549.00
1126  Lenovo  2 in 1 Convertible    10.1       479.00
1082  Lenovo  2 in 1 Convertible    10.1       646.27
50    Lenovo  2 in 1 Convertible    10.1       319.00

In [41]:
df[['Company', 'TypeName', 'Inches', 'Price_euros']][df['Inches'] == df['Inches'].min()]

Company            TypeName  Inches  Price_euros
50    Lenovo  2 in 1 Convertible    10.1       319.00
1082  Lenovo  2 in 1 Convertible    10.1       646.27
1114  Lenovo  2 in 1 Convertible    10.1       549.00
1126  Lenovo  2 in 1 Convertible    10.1       479.00

### 4.2 Сколько стоит самый дорогой ноутбук у компании HP?

In [44]:
df2 = df[['Company', 'TypeName', 'Inches', 'Price_euros']][df['Company'] == 'HP']
df2

Company   TypeName  Inches  Price_euros
2         HP   Notebook    15.6       575.00
10        HP   Notebook    15.6       393.90
11        HP   Notebook    15.6       344.99
22        HP   Notebook    15.6       258.00
24        HP  Ultrabook    15.6       659.00
...      ...        ...     ...          ...
1282      HP    Netbook    11.6       209.00
1287      HP   Notebook    15.6       764.00
1294      HP   Notebook    15.6       549.99
1296      HP    Netbook    11.6       209.00
1301      HP   Notebook    15.6       764.00

[274 rows x 4 columns]

In [45]:
df2[df2['Price_euros'] == df2['Price_euros'].max()]

Company     TypeName  Inches  Price_euros
749      HP  Workstation    17.3       4389.0

### 4.3 Как много ноутбуков Ultrabook с 8GB RAM?
Найдите сколько таких ультрабуков с 8GB ОЗУ в процентном соотношении
относительно всех ультрабуков

In [46]:
df_ultras =df[df['TypeName'] == 'Ultrabook']
df_ultras

laptop_ID Company               Product   TypeName  Inches  \
0             1   Apple           MacBook Pro  Ultrabook    13.3   
1             2   Apple           Macbook Air  Ultrabook    13.3   
3             4   Apple           MacBook Pro  Ultrabook    15.4   
4             5   Apple           MacBook Pro  Ultrabook    13.3   
6             7   Apple           MacBook Pro  Ultrabook    15.4   
...         ...     ...                   ...        ...     ...   
1236       1254      HP         EliteBook 840  Ultrabook    14.0   
1241       1259    Asus  ZenBook UX310UA-WB71  Ultrabook    13.3   
1261       1279    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   
1275       1293    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   
1289       1307    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   

                        ScreenResolution                         Cpu   Ram  \
0     IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                               1440x900        Intel Core i5 1.8GHz   8GB   
3     IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4     IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   
6     IPS Panel Retina Display 2880x1800        Intel Core i7 2.2GHz  16GB   
...                                  ...                         ...   ...   
1236                           2560x1440  Intel Core i7 6500U 2.5GHz   8GB   
1241                   Full HD 1920x1080  Intel Core i7 6500U 2.5GHz   8GB   
1261         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz   8GB   
1275         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz   8GB   
1289         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz   8GB   

                   Memory                           Gpu       OpSys  Weight  \
0               128GB SSD  Intel Iris Plus Graphics 640       macOS  1.37kg   
1     128GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
3               512GB SSD            AMD Radeon Pro 455       macOS  1.83kg   
4               256GB SSD  Intel Iris Plus Graphics 650       macOS  1.37kg   
6     256GB Flash Storage       Intel Iris Pro Graphics    Mac OS X  2.04kg   
...                   ...                           ...         ...     ...   
1236            256GB SSD         Intel HD Graphics 520   Windows 7  1.54kg   
1241            256GB SSD         Intel HD Graphics 520  Windows 10  1.45kg   
1261            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   
1275            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   
1289            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   

      Price_euros  
0         1339.69  
1          898.94  
3         2537.45  
4         1803.60  
6         2139.97  
...           ...  
1236      2198.19  
1241      1280.00  
1261       729.00  
1275       729.00  
1289       729.00  

[196 rows x 13 columns]

In [47]:
df_ultras_8 =df_ultras[df_ultras['Ram'] == '8GB']
df_ultras_8.shape[0]

df_ultras_8.shape[0]/df_ultras.shape[0]

0.7142857142857143

## Задание 5.

### 5.1 Выберите ноутбук клиенту
Клиент хочет подобрать ноутбук с 8GB или 16GB ОЗУ на Windows 10 в
стоимости до 500 евро, сколько у него вариантов?

In [48]:
df_by_price =df[df['Price_euros'] <= 500]
df_by_price

laptop_ID Company                              Product  TypeName  \
5             6    Acer                             Aspire 3  Notebook   
10           11      HP                               250 G6  Notebook   
11           12      HP                               250 G6  Notebook   
13           14    Dell                        Inspiron 3567  Notebook   
18           19  Lenovo                    IdeaPad 320-15IKB  Notebook   
...         ...     ...                                  ...       ...   
1291       1309    Acer                       Aspire ES1-531  Notebook   
1292       1310    Dell                        Inspiron 3552  Notebook   
1296       1314      HP                     Stream 11-Y000na   Netbook   
1300       1318  Lenovo                   IdeaPad 100S-14IBR  Notebook   
1302       1320    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)  Notebook   

      Inches   ScreenResolution                                   Cpu  Ram  \
5       15.6           1366x768               AMD A9-Series 9420 3GHz  4GB   
10      15.6           1366x768            Intel Core i5 7200U 2.5GHz  4GB   
11      15.6  Full HD 1920x1080              Intel Core i3 6006U 2GHz  4GB   
13      15.6  Full HD 1920x1080              Intel Core i3 6006U 2GHz  4GB   
18      15.6  Full HD 1920x1080            Intel Core i3 7100U 2.4GHz  8GB   
...      ...                ...                                   ...  ...   
1291    15.6           1366x768  Intel Celeron Dual Core N3060 1.6GHz  4GB   
1292    15.6           1366x768  Intel Celeron Dual Core N3050 1.6GHz  2GB   
1296    11.6           1366x768  Intel Celeron Dual Core N3060 1.6GHz  2GB   
1300    14.0           1366x768  Intel Celeron Dual Core N3050 1.6GHz  2GB   
1302    15.6           1366x768  Intel Celeron Dual Core N3050 1.6GHz  4GB   

                  Memory                    Gpu       OpSys  Weight  \
5              500GB HDD          AMD Radeon R5  Windows 10   2.1kg   
10             500GB HDD  Intel HD Graphics 620       No OS  1.86kg   
11             500GB HDD  Intel HD Graphics 520       No OS  1.86kg   
13             256GB SSD     AMD Radeon R5 M430  Windows 10   2.2kg   
18               1TB HDD   Nvidia GeForce 940MX       No OS   2.2kg   
...                  ...                    ...         ...     ...   
1291           500GB HDD  Intel HD Graphics 400       Linux   2.4kg   
1292           500GB HDD      Intel HD Graphics  Windows 10  2.20kg   
1296  32GB Flash Storage  Intel HD Graphics 400  Windows 10  1.17kg   
1300  64GB Flash Storage      Intel HD Graphics  Windows 10   1.5kg   
1302           500GB HDD      Intel HD Graphics  Windows 10   2.2kg   

      Price_euros  
5          400.00  
10         393.90  
11         344.99  
13         498.90  
18         499.00  
...           ...  
1291       289.00  
1292       379.00  
1296       209.00  
1300       229.00  
1302       369.00  

[246 rows x 13 columns]

In [50]:
df_by_OS = df_by_price[df_by_price['OpSys'] == 'Windows 10']
df_by_OS

laptop_ID Company                              Product  TypeName  \
5             6    Acer                             Aspire 3  Notebook   
13           14    Dell                        Inspiron 3567  Notebook   
20           21    Asus                      Vivobook E200HA   Netbook   
25           26    Dell                        Inspiron 3567  Notebook   
30           31   Chuwi                        LapBook 15.6"  Notebook   
...         ...     ...                                  ...       ...   
1290       1308    Dell                        Inspiron 3567  Notebook   
1292       1310    Dell                        Inspiron 3552  Notebook   
1296       1314      HP                     Stream 11-Y000na   Netbook   
1300       1318  Lenovo                   IdeaPad 100S-14IBR  Notebook   
1302       1320    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)  Notebook   

      Inches   ScreenResolution                                   Cpu  Ram  \
5       15.6           1366x768               AMD A9-Series 9420 3GHz  4GB   
13      15.6  Full HD 1920x1080              Intel Core i3 6006U 2GHz  4GB   
20      11.6           1366x768           Intel Atom x5-Z8350 1.44GHz  2GB   
25      15.6           1366x768              Intel Core i3 6006U 2GHz  4GB   
30      15.6  Full HD 1920x1080           Intel Atom x5-Z8300 1.44GHz  4GB   
...      ...                ...                                   ...  ...   
1290    15.6           1366x768            Intel Core i3 7100U 2.4GHz  6GB   
1292    15.6           1366x768  Intel Celeron Dual Core N3050 1.6GHz  2GB   
1296    11.6           1366x768  Intel Celeron Dual Core N3060 1.6GHz  2GB   
1300    14.0           1366x768  Intel Celeron Dual Core N3050 1.6GHz  2GB   
1302    15.6           1366x768  Intel Celeron Dual Core N3050 1.6GHz  4GB   

                  Memory                    Gpu       OpSys  Weight  \
5              500GB HDD          AMD Radeon R5  Windows 10   2.1kg   
13             256GB SSD     AMD Radeon R5 M430  Windows 10   2.2kg   
20    32GB Flash Storage  Intel HD Graphics 400  Windows 10  0.98kg   
25               1TB HDD  Intel HD Graphics 520  Windows 10   2.3kg   
30    64GB Flash Storage      Intel HD Graphics  Windows 10  1.89kg   
...                  ...                    ...         ...     ...   
1290             1TB HDD  Intel HD Graphics 620  Windows 10   2.3kg   
1292           500GB HDD      Intel HD Graphics  Windows 10  2.20kg   
1296  32GB Flash Storage  Intel HD Graphics 400  Windows 10  1.17kg   
1300  64GB Flash Storage      Intel HD Graphics  Windows 10   1.5kg   
1302           500GB HDD      Intel HD Graphics  Windows 10   2.2kg   

      Price_euros  
5          400.00  
13         498.90  
20         191.90  
25         418.64  
30         244.99  
...           ...  
1290       459.00  
1292       379.00  
1296       209.00  
1300       229.00  
1302       369.00  

[165 rows x 13 columns]

In [53]:
df_end = df_by_OS[(df_by_OS['Ram'] == '8GB') | (df_by_OS['Ram'] == '16GB')] 
df_end.shape[0]

13

In [55]:
df_end2 =df[(df['Price_euros'] <= 500) & (df['OpSys'] == 'Windows 10') & ((df['Ram'] == '8GB') | (df['Ram'] == '16GB'))]
df_end2.shape[0]

13

### 5.2 Выберите ноутбук клиенту
Клиент хочет подобрать ноутбук от MSI, с видеокартой Nvidia GeForce GTX
1050 Ti и главное не с диагональю 15.6. В какой ценовой категории вышли
подобные ноутбуки?

### 5.3 Что дешевле?
В среднем дешевле ноутбуки с CPU Intel Core i7 7700HQ 2.8GHz или с Intel
Core i7 7600U 2.8GHz?

## Задание 6.
Найдите самый легкий ноутбук
Но обратите внимание на тип и представление данных в признаке
Weight, если что, замените в строке 'kg' на пустую строку через
метод .str.replace()